src/cdn.rs

```diff
@@ -116,6 +117,18 @@ impl CdnBackend {
     }
 }
 
+pub(crate) fn invalidate_crate(config: &Config, cdn: &CdnBackend, name: &str) -> Result<()> {
+    if let Some(distribution_id) = config.cloudfront_distribution_id_web.as_ref() {
+        cdn.create_invalidation(
+            distribution_id,
+            &[&format!("/{}*", name), &format!("/crate/{}*", name)],
```

Comment:
One thing I just realised, when invalidating `tokio` this will also invalidate `tokio-utils`. Would it be worth using `["/{name}", "/{name}/*", "/crate/{name}", "/crate/{name}/*"]` to avoid that?

Reply:
My reading of those docs is that a `*` is treated essentially as a `.*` on a regex of the full path, they don't treat `/` specially in any way. (I really hate example driven documentation, the first two examples are either implying that `*` is really magical, or there's a lot of unstated assumptions about the examples that explains why they have different behavior).

Reply:
I was also confused by this, and read the same documentation @syphar read, and came to the same conclusions.

However, @Nemo157's comment prompted me to re-read the docs and I've now concluded the docs are ambiguous / misleading, and @Nemo157's interpretation is very likely right.

> To invalidate all of the files in a directory:
> /directory-path/*
> To invalidate a directory, all of its subdirectories, and all of the files in the directory and subdirectories:
> /directory-path*

These two lines, contrasted, suggest that there is some very surprising magic in place such that the first example matches only _files_, not directories and the latter matches files and directories, recursively. However, contrast with the last example:

> To invalidate all of the files in a distribution:
> /*

This seems to indicate the recursive semantics. I think the first example is just misleadingly described.

A secondary source seems to back this up: https://aws.amazon.com/about-aws/whats-new/2015/05/amazon-cloudfront-makes-it-easier-to-invalidate-multiple-objects/

Reply:
I'll do some testing around this with my dummy distribution